docs/ghc-deprecation.md

```diff
@@ -0,0 +1,39 @@
+# GHC support deprecation policy
+
+- `haskell-language-server`(HLS) is highly tied to ghc api so much so that it needs a specific flavour for each ghc minor version to ensure it will work in a reliable way
+- It supposes the codebase is riddled with code conditioned to each supported ghc versions.  
+  - It even needs entire packages to fully support older versions of ghc.
```

Comment:
I think this is a prose masquerading as a bullet list :)

Reply:
```suggestion
```